src/char/int_storage.c

```diff
@@ -41,24 +41,27 @@ struct inter_storage_interface inter_storage_s;
 struct inter_storage_interface *inter_storage;
 
 /// Save storage data to sql
-int inter_storage_tosql(int account_id, struct storage_data* p)
+int inter_storage_tosql(int account_id, const struct storage_data *p)
 {
 	nullpo_ret(p);
-	chr->memitemdata_to_sql(p->items, MAX_STORAGE, account_id, TABLE_STORAGE);
+
+	chr->memitemdata_to_sql(VECTOR_DATA(*p), VECTOR_LENGTH(*p), account_id, TABLE_STORAGE);
+
 	return 0;
 }
 
 /// Load storage data to mem
-int inter_storage_fromsql(int account_id, struct storage_data* p)
+int inter_storage_fromsql(int account_id, struct storage_data *p)
 {
 	StringBuf buf;
 	char* data;
 	int i;
 	int j;
+	int num_rows = 0;
 
 	nullpo_ret(p);
-	memset(p, 0, sizeof(struct storage_data)); //clean up memory
-	p->storage_amount = 0;
+
+	VECTOR_INIT(*p);
```

Comment:
`VECTOR_INIT()` should be on the caller's side (so that you don't have to pass an uninitialized variable to a function), and here you should have `VECTOR_CLEAR()` (just as a safety measure). This change would make the  caller the owner of this struct, which is what we want here so that allocations and deallocations are always in matching pair where the owner resides